### ISSUE TITLE
Preselection & Selection Colors

### ISSUE THREAD

**Issue by ickby** [WandererFan/FreeCAD-TechDraw#60]
Preselection color and selection color do not match the preference settings. Preselection is black (and hence not visible when a black edge is preselected) and selection color is blue, while TechDraw preferences defaults say yellow and green.  Chaning the setting does not effect the colors in TechDraw.

IMHO it woul dmake sense to not have a custom color preference but to use the generall frecad preselection and selection color user setting also for tech draw.

OS: Windows 7
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.17.7965 (Git)
Build type: Release
Branch: master
Hash: c5b501b9d3ccb538f0a075a59be8dd0210c62ba9
Python version: 2.7.8
Qt version: 4.8.7
Coin version: 4.0.0a
OCC version: 7.0.0

**Comment by WandererFan** [comment]
I can't reproduce this here - I get black/green/yellow and I can change with preferences.  Coulld you please send me this section of your usr.cfg:

```
    <FCParamGroup Name="Mod">
      <FCParamGroup Name="TechDraw">
        <FCParamGroup Name="Colors">
          <FCUInt Name="NormalColor" Value="421075455"/>
          <FCUInt Name="SelectColor" Value="481107199"/>
          <FCUInt Name="PreSelectColor" Value="4278190335"/>  
```

The reason for a separate colour choice is that some people use odd backgrounds in the 3D view and they might need a different set against the white of the drawing page background.

**Comment by WandererFan** [comment]
Typo in default preselect if no pref set:  0x00080800  sb 0x00808000  explains black preselect,
QGIPrimPath.cpp and QGIView.cpp

**Comment by WandererFan** [comment]
- PrimPath/QGIView
  - default select is missing setpackedvalue
  - change of preferences is not reflected in objects previously created
    - works in newly created objects or after document close/reopen

**Comment by ickby** [comment]
Ah yes, if a view is recomputed it takes the new color values. This seems very unintuitive. There is a callback when preferences change, I think in the ViewProvider,, maybe looking out for the selection colors there is a useful thing

**Comment by WandererFan** [comment]
fixed in PR "Color Preferences #218".  Still need to wait for object repaint to see result.  

Need a "repaint" button??
